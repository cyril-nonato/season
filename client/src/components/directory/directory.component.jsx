import React from 'react'
import DirectoryItem from '../directory-item/directory-item.component';

import * as S from './directory.styles';

const Directory = ({sections}) => {
  return ( 
    <S.Container>
      {
        sections.map(section => <DirectoryItem key={section.id} item={section} />)
      }
    </S.Container>
   );
}
 
export default Directory;