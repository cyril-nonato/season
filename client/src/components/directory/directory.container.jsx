import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';
import { compose } from 'redux';

import { selectSections } from '../../redux/home/home.selector';
import Directory from './directory.component';

const mapStateToProps = createStructuredSelector({
  sections: selectSections
});

const DirectoryContainer = compose(connect(mapStateToProps))(Directory);

export default DirectoryContainer;