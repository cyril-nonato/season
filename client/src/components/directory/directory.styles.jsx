import styled from 'styled-components';
import { mediaQueries } from '../../theme/theme';

export const Container = styled.div`
  width: 100%;
  margin: 0 auto;
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-gap: 2rem;
  margin-bottom: 5rem;

  @media only screen and (max-width: ${mediaQueries.windows}) {
    grid-template-columns: 1fr 1fr 1fr;
  };

  @media only screen and (max-width: ${mediaQueries.tablet}) {
    grid-template-columns: 1fr 1fr;
  };

  @media only screen and (max-width: ${mediaQueries.phone}) {
    grid-template-columns: 1fr;
  }
`;