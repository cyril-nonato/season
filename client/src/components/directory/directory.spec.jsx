import React from 'react'
import {cleanup, render} from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import 'jest-styled-components'

import * as S from './directory.styles'

jest.mock('../directory-item/directory-item.component', () => {
  return (
    <div data-testid='directory-item'>
      Directory Item
    </div>
  )
});

afterEach(cleanup);

describe('Directory styles Test', () => {
  
  test('Should match the snapspshot', () => {
    const {asFragment} = render(<S.Container />);

    expect(asFragment()).toMatchSnapshot();
  })

});