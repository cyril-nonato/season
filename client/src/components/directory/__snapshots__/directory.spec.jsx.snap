// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory styles Test Should match the snapspshot 1`] = `
<DocumentFragment>
  .c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0 auto;
}

@media only screen and (min-width:700px) {
  .c0 {
    display: grid;
    grid-template-columns: 1fr 1fr;
    grid-gap: 5rem;
  }
}

<div
    class="c0"
  />
</DocumentFragment>
`;
